### PR TITLE
Annotations generate a name collision - fixed

### PR DESCRIPTION
**Pull request by davidhubbard**
This is a partial fix of https://github.com/capnproto/go-capnp/issues/46

> zombiezen commented on Aug 22, 2016:
>
> One of the notable examples is in persistent.capnp that the annotation and
> the type collide.

Also see https://github.com/capnproto/capnproto/issues/323

This commit changes the generated code for all annotations that start with a lowercase letter. Since the generated code in Go must start with an uppercase letter to be valid Go, this change appends an "_" to the name, regardless of whether there is a name collision already. (The thinking is that a name collision can happen in the future, and the renaming must be strictly an injective function, meaning that Go identifiers are always chosen the same regardless of other names. This way the codegen will not cause painful refactors in other code due to a name change later.)

The capnp files in this repo under `std` all have $Go.name() annotations due to this bug. This removes one of them (std/capnp/persistent.capnp) and replaces it with a unit test that compiles the generated code. Actually doing a "go build" is one step father than previous unit tests.

The thinking is to not add too many calls to "go build," since there is a lot of overhead and slow unit tests are discouraging. This commit appears to add about 250ms to the time it takes to run a "go test" in the capnpc-go subdir.

### CODE REVIEW

**Comment by davidhubbard** [comment]
**Draft pull request**

`go test` in the capnpc-go dir still fails with: `./persistent-simple.capnp.out.go:49:33: undefined: context`

**Comment by davidhubbard** [comment]
`level0_test.go:2270: log level error: rpc: incoming disembargo: send receiver loopback: send message: rpc: build message: target for receiver loopback does not point to the right connection (args: map[])    senderpromise_test.go:404: Received unexpected error: test.capnp:PingPong.echoNum: call on null client`

The current CI status already had that error. This PR doesn't address it, but at least it wasn't caused here. See [previous CI builds](https://github.com/capnproto/go-capnp/actions/runs/5443927637)

File: capnpc-go/capnpc-go_test.go [hunk 1]
```diff
@@ -467,3 +472,144 @@ func nodeListString(n []*node) string {
...
+// * src/capnp/persistent.capnp
+// * Also found in this repo: std/capnp/persistent.capnp
+//
+// It contains two definitions:
+//   interface Persistent {}
+//   annotation persistent(interface, field) :Void;
+//
+// testdata/persistent-simple.capnp is a minimal reproducible test case for the
+// collision.
+func TestPersistent(t *testing.T) {
```
Comment (lthibault) [inline capnpc-go/capnpc-go_test.go:514]: If possible/reasonable, I would recommend running `t.Parallel()` here.  We have a fairly large test suite, and the parallelism really helps.

**Review by lthibault — CHANGES_REQUESTED** [review]
This is just a first pass to pick out things that jump out.  Digging in more deeply now.

Looks great so far! 👍

**Review by lthibault — APPROVED** [review]
This is easily one of the best PRs I've ever seen!  Thanks for your contribution -- approved! 🎉

File: capnpc-go/capnpc-go_test.go (hunk 1 above)
Comment (davidhubbard) [reply capnpc-go/capnpc-go_test.go:514]: Good idea, I added t.Parallle() at `capnpac-go_test.go:525`

**Review by lthibault — APPROVED** [review]

**Pull request merged**